### DISCUSSION TITLE
Complex Types

### DISCUSSION THREAD

**Discussion by shanerbaner82** [CodeDredd/laravel-soap#232] [Q&A]
How does one go about creating a complex type to insert into a call? I have one call that accepts a single complex type, but some of the objects inside of that type are also complex types, it goes down like that 3-4 levels or so.

Is this documented anywhere?

**Comment by CodeDredd** [comment]
Hey,

no its not. The documentation for v3 is still lacking. Sorry for that. So far v2 docs is still there.
But i don't know what you mean with "complex types". Can you give me an example?

**Reply by shanerbaner82** [reply]
I am working off a WSDL that has the structure [seen here](https://gist.github.com/a2b160c64ba1176921c47abaa217fe2d). You can see `<s:element>`s that contain elements that refer to a defined 'type' below. How would I structure this using this package?
